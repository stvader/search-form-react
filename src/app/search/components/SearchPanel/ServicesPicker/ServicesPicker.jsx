import React from 'react';
import { string, func } from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Autocomplete from '@material-ui/lab/Autocomplete';

import { servisesList } from '../../../search.servicesList';

const ServicesPicker = ({ selectService, handleServiceChange }) => (
  <Autocomplete
    options={servisesList}
    groupBy={option => option.category}
    getOptionLabel={option => option.title}
    onChange={(e, val) => handleServiceChange(val && val.title)}
    value={servisesList.find(({ title }) => title === selectService)}
    renderInput={params => (
      <TextField {...params} label="Pleace select service" variant="outlined" fullWidth />
    )}
  />
);

ServicesPicker.propTypes = {
  selectService: string,
  handleServiceChange: func,
};

export default ServicesPicker;
