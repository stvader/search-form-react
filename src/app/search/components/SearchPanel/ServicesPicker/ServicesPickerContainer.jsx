import React, { memo } from 'react';

import { SearchPanelContext } from '../SearchPanelContext';

import ServicesPicker from './ServicesPicker';

import { useAutomplete } from '../../../../common/hooks/useAutocomplete';

const ServicesPickerContainer = () => {
  const { value, handleSelect } = useAutomplete(SearchPanelContext);

  return <ServicesPicker selectService={value} handleServiceChange={handleSelect} />;
};

export default memo(ServicesPickerContainer);
