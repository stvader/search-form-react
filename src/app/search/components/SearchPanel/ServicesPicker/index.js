import ServicesPicker from './ServicesPickerContainer';

export default ServicesPicker;
