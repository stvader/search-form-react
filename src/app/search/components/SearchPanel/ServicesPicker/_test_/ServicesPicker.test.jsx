import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Autocomplete from '@material-ui/lab/Autocomplete';

import ServicesPicker from '../ServicesPicker';

configure({ adapter: new Adapter() });

const mockHandleChange = jest.fn();

describe('ServicesPicker', () => {
  it('ServicesPicker render', () => {
    const component = shallow(
      <ServicesPicker selectService="Some service" onChange={mockHandleChange} />,
    );

    const input = component.find(Autocomplete);

    expect(component).toMatchSnapshot();
    expect(input).toBeDefined();
  });
});
