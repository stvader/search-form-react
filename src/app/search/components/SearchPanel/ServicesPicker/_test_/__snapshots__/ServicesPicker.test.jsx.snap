// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesPicker ServicesPicker render 1`] = `ShallowWrapper {}`;
